v8/Client/ParallelServiceProxy.cs

```diff
@@ -971,15 +981,41 @@ public IEnumerable<OrganizationResponse> Execute(IEnumerable<OrganizationRequest
             // Handle faults
             if (errorHandler != null)
             {
-                foreach(var failure in allFailures)
+                foreach (var failure in allFailures)
                 {
-                    errorHandler(failure.Request, failure.Exception);                  
+                    errorHandler(failure.Request, failure.Exception);
                 }
             }
 
             return allResponses;
         }
 
+
+        private static void ApplyDelay(FaultException<OrganizationServiceFault> e, int retryCount)
+        {
+            TimeSpan delay;
+            if (e.Detail.ErrorCode == RateLimitExceededErrorCode)
+            {
+                // Use Retry-After delay when specified
+                delay = (TimeSpan)e.Detail.ErrorDetails["Retry-After"];

```

Comment:
Need to add a check here to see if "Retry-After" exists in the ErrorDetails, as this isn't always the case. There are different types of RateLimits represented by the same RateLimitExceededErrorCode, so don't assume. I already ran into this issue.